modules/aruco/src/aruco.cpp

```diff
@@ -439,25 +453,19 @@ static void _detectInitialCandidates(const Mat &grey, vector< vector< Point2f >
 /**
  * @brief Detect square candidates in the input image
  */
-static void _detectCandidates(InputArray _image, vector< vector< vector< Point2f > > >& candidatesSetOut,
+static void _detectCandidates(InputArray _grayImage, vector< vector< vector< Point2f > > >& candidatesSetOut,
                               vector< vector< vector< Point > > >& contoursSetOut, const Ptr<DetectorParameters> &_params) {
 
-    Mat image = _image.getMat();
-    CV_Assert(image.total() != 0);
-
-    /// 1. CONVERT TO GRAY
-    Mat grey;
-    _convertToGrey(image, grey);
```

Comment:
input image was gray already

Reply:
Add checks about assumptions of this code:
- not `.empty()`
- `CV_CheckType()`

Reply:
these conditions were checked already (in detectMarkers)

added `CV_DbgAssert`